deployments/pre-publish.js

```diff
@@ -122,6 +93,21 @@ const main = async () => {
   console.log('Generating exports');
   generateExports();
 
+  console.log('Building source');
+  await build({
+    entry: ['deployments/src/index.ts'],
+    outDir: 'deployments/dist',
+    format: ['cjs', 'esm'],
+    splitting: false,
+    sourcemap: true,
+    clean: true,
+    dts: true,
+    publicDir: 'generated', // copy generated files to dist
```

Comment:
Is `generated` a temporary folder?

Reply:
Yes, it is used to store generated files that are used during compilation. So we build files (like `abis.ts` now) manually and then feed it to `tsup` for compilation. That way we can import these files before they exist like we do [here](https://github.com/NexusMutual/smart-contracts/pull/913/files#diff-e5ae286dd52131f050edbe32b9fcab68d602f076da1a755e8e2a71778cb66124R2).